pause.go

```diff
@@ -14,18 +14,36 @@ package main
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <errno.h>
 
 #define PAUSE_BIN "pause-bin"
 
-void __attribute__((constructor)) sandbox_pause(int argc, const char **argv) {
-	if (argc != 2 || strcmp(argv[1], PAUSE_BIN)) {
-		return;
-	}
+void __attribute__((constructor)) sandbox_pause() {
+        char buf[512], *p;
+        int fd, len;
 
-	for (;;) pause();
+        fd = open("/proc/self/cmdline", O_RDONLY);
+        if (fd < 0) {
+                fprintf(stderr, "failed to open proc: %d\n", errno);
+                return;
```

Comment:
IMO, we should exit here, otherwise, we got two kata-agent processes running in parallel.

Reply:
I'd prefer to use `strerror(errno)` to be a more verbose

Reply:
Good point! I'll update.